dependencies/clients/teku_validator.go

```diff
@@ -240,7 +240,8 @@ func (t *TekuValidatorClient) List(ctx *cli.Context) (err error) {
 			return nil
 		}
 
-		if !strings.Contains(d.Name(), "json") {
+		keystoreExt := filepath.Ext(d.Name())
+		if !strings.Contains(keystoreExt, "json") {
```

Comment:
Minor improvement. Checking the file name and extension seems sufficient to throw an error. 

However, consider reading and parsing the JSON content if the file or error is used within any logic.

```
if err := json.Unmarshal(content, &jsonData); err != nil {
		return false
	}
```

Reply:
> Checking the file name and extension seems sufficient to throw an error.
> 
> However, consider reading and parsing the JSON content if the file or error is used within any logic.
> 
> ```
> if err := json.Unmarshal(content, &jsonData); err != nil {
> 		return false
> 	}
> ```

Checking for unmarshaling error is done few lines below - what could actually be done is wrapping the error for a message that would not stop the code execution. But we have control of what is in the keystore folder, so as you said - just checking for extension is I think sufficient here